.github/workflows/terratest.yml

```diff
@@ -12,11 +12,21 @@ jobs:
   go-tests:
     name: Run Go Tests
     runs-on: ubuntu-latest
-    timeout-minutes: 30
+    timeout-minutes: 0
+    strategy:
+      fail-fast: false
+      matrix:
+        network: ["dev", "qa", "test", "main"]
+        include: 
+          - describe: "git describe --match 'v[0-9]*.[0-9]*.[0-9]*'  --exclude='*-rc*' --exclude='*-qa*' --abbrev=0 HEAD"
+            network: "dev"
+          - describe: "git describe --match 'v[0-9]*.[0-9]*.[0-9]*'  --exclude='*-rc*' --exclude='*-dev*' --abbrev=0 HEAD"
+            network: "qa"
```

Comment:
i think we should exclude the main version somehow from those 3 elements because if we have the latest main branch 2.8.0 and rc 2.8.0-rc and qa 2.8.0-qa the above lines will all match the main